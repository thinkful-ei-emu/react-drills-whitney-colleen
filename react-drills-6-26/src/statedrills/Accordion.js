import React from 'react';

class Accordion extends React.Component {

  constructor(props) {
    super(props)
    this.state = {
      isShowing: null
    }
  }
  static defaultProps = {
    sections: []
  }
  
  handleClick = (event) => {
    this.setState({
      isShowing: +event.currentTarget.id
    })
    console.log(event.currentTarget);
  }


  render() {
    let buttonList = this.props.sections.map((section, index) => {
      console.log(index, this.state.isShowing)
      return (
        <li key={index} id={index} onClick={this.handleClick}>
          <button >{section.title}</button>
          {this.state.isShowing === index && <p>{section.content}</p>}
        </li>
      )
    })

    return(
      <ul>
        {buttonList}
      </ul>
    );
  }
}

export default Accordion;