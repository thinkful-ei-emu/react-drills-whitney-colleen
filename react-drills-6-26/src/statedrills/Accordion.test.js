import { shallow } from 'enzyme';
import { configure } from 'enzyme';
import toJson from 'enzyme-to-json';
import Adapter from 'enzyme-adapter-react-16';
import React from 'react';
import ReactDOM from 'react-dom';
import Accordion from './Accordion';

configure({ adapter: new Adapter() })

describe('Accordion Component', () =>{
// The component renders an empty li when the sections prop is not supplied
// The component renders no sections as active by default
// The component opens a clicked section
// The component only opens the last section when multiple sections have been clicked.

const sections = [{
  title: 'Section 1',
  content: 'Lorem ipsum dolor sit amet consectetur adipisicing elit.',
},
{
  title: 'Section 2',
  content: 'Cupiditate tenetur aliquam necessitatibus id distinctio quas nihil ipsam nisi modi!',
},
{
  title: 'Section 3',
  content: 'Animi amet cumque sint cupiditate officia ab voluptatibus libero optio et?',
}]

it('component renders an empty li when sections prop is not supplied', () => {
  const wrapper = shallow(<Accordion />);
  expect(toJson(wrapper)).toMatchSnapshot();
})

it('component renders no sections as active by default', () => {
  const wrapper = shallow(<Accordion sections={sections}/>)
  expect(toJson(wrapper)).toMatchSnapshot();
})

it('component opens a clicked section', () => {
  const wrapper = shallow(<Accordion sections={sections} />)
  wrapper.find('li').at(0).simulate('click', {currentTarget: 'li'})
  expect(toJson(wrapper)).toMatchSnapshot()
})
it('component opens last section if multiple are clicked', () => {
  const wrapper = shallow(<Accordion sections={sections} />)
  wrapper.find('li').at(0).simulate('click', {currentTarget: 'li'})
  wrapper.find('li').at(2).simulate('click', {currentTarget: 'li'})
  expect(toJson(wrapper)).toMatchSnapshot()
})
})