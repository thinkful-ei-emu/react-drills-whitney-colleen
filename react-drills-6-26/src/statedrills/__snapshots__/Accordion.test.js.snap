// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Accordion Component component opens a clicked section 1`] = `
<ul>
  <li
    id={0}
    key="0"
    onClick={[Function]}
  >
    <button>
      Section 1
    </button>
  </li>
  <li
    id={1}
    key="1"
    onClick={[Function]}
  >
    <button>
      Section 2
    </button>
  </li>
  <li
    id={2}
    key="2"
    onClick={[Function]}
  >
    <button>
      Section 3
    </button>
  </li>
</ul>
`;

exports[`Accordion Component component opens last section if multiple are clicked 1`] = `
<ul>
  <li
    id={0}
    key="0"
    onClick={[Function]}
  >
    <button>
      Section 1
    </button>
  </li>
  <li
    id={1}
    key="1"
    onClick={[Function]}
  >
    <button>
      Section 2
    </button>
  </li>
  <li
    id={2}
    key="2"
    onClick={[Function]}
  >
    <button>
      Section 3
    </button>
  </li>
</ul>
`;

exports[`Accordion Component component renders an empty li when sections prop is not supplied 1`] = `<ul />`;

exports[`Accordion Component component renders no sections as active by default 1`] = `
<ul>
  <li
    id={0}
    key="0"
    onClick={[Function]}
  >
    <button>
      Section 1
    </button>
  </li>
  <li
    id={1}
    key="1"
    onClick={[Function]}
  >
    <button>
      Section 2
    </button>
  </li>
  <li
    id={2}
    key="2"
    onClick={[Function]}
  >
    <button>
      Section 3
    </button>
  </li>
</ul>
`;

exports[`component opens a clicked section 1`] = `<ul />`;

exports[`component renders an empty li when sections prop is not supplied 1`] = `<ul />`;

exports[`component renders no sections as active by default 1`] = `
<ul>
  <li
    id={0}
    onClick={[Function]}
  >
    <button>
      Section 1
    </button>
  </li>
  <li
    id={1}
    onClick={[Function]}
  >
    <button>
      Section 2
    </button>
  </li>
  <li
    id={2}
    onClick={[Function]}
  >
    <button>
      Section 3
    </button>
  </li>
</ul>
`;
